homeassistant/components/hardware/websocket_api.py

```diff
@@ -79,12 +78,15 @@ async def ws_subscribe_system_status(
 
     system_status: SystemStatus = hass.data[DOMAIN]["system_status"]
 
-    @callback
-    def async_update_status(now: datetime) -> None:
+    async def async_update_status(now: datetime) -> None:
         cpu_percentage = round(
-            system_status.ha_psutil.psutil.cpu_percent(interval=None)
+            await hass.async_add_executor_job(
+                system_status.ha_psutil.psutil.cpu_percent
+            )
+        )
+        virtual_memory = await hass.async_add_executor_job(
```

Comment:
It's better to wrap all the sync calls inside a function and schedule that once on the executor.